### PR TITLE
Propose maintinaing a contributors.md file to manage access changes

### PR DESCRIPTION
**Pull request by goofballLogic**

### CODE REVIEW

**Comment by asbjornu** [comment]
This is a nice addition, but perhaps we can add it as the special [`CODEOWNERS`](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) file instead, as that is understood by GitHub and serves multiple purposes?

**Comment by goofballLogic** [comment]
I think CODEOWNERS can only contain default reviewers (which we also need)?

I was hoping for a way manage suggesting new contributors via consensus. For example, yesterday I added @wattsm based on my personal knowledge of Mark's prowess with .Net, but there wasn't an easy way for me to run this past you for agreement. This file would allow me to do a PR with his name, and ask for your sign off before proceeding.

Or, I could be overthinking it....

File: contributors.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+## Admins
+
+ - Asbjørn Ulsberg https://github.com/asbjornu 
+ - Andrew Stewart Gibson https://github.com/goofballLogic
+ 
+## Contributors
+
+ - Tomasz Pluskiewicz https://github.com/tpluscode
+ - Ben Abelshausen https://github.com/xivk
```
Comment (asbjornu) [inline contributors.md:15]:
```suggestion
# Contributors and maintainers

We maintain this list as a way to track who's currently involved in the project.
Please use PRs to this document to propose adding new maintainers /
contributors.

**N.B.** please propose names ordered alphabetically by username.

## Admins

 - [Asbjørn Ulsberg](https://github.com/asbjornu )
 - [Andrew Stewart Gibson](https://github.com/goofballLogic)
 
## Contributors

 - [Tomasz Pluskiewicz](https://github.com/tpluscode)
 - [Ben Abelshausen](https://github.com/xivk)
```

File: contributors.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Contributors and maintainers
+
+We maintain this list as a way to track who's currently involved in the project.
+Please use PRs to this document to propose adding new maintainers /
+contributors.
+
+<a href="https://github.com/linked-data-dotnet/json-ld.net/graphs/contributors">
+  <img src="https://contributors-img.web.app/image?repo=linked-data-dotnet/json-ld.net" />
+</a>
```
Comment (asbjornu) [inline contributors.md:9]:
```suggestion
[![Contributors](https://contributors-img.web.app/image?repo=linked-data-dotnet/json-ld.net)](https://github.com/linked-data-dotnet/json-ld.net/graphs/contributors)
```

**Comment by goofballLogic** [comment]
@asbjornu do you know why we don't have Squash option enabled for this repo?

**Comment by asbjornu** [comment]
@goofballLogic, I've disabled it. I'm very strongly against everything GitHub offers in terms of updating branches, squashing and merging. Everything they do and enable in the UI, I basically like the opposite. 😄 

Let's take [squash merge](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges#squash-and-merge-your-pull-request-commits) as an example. The biggest problem with that is this:

>Pull requests with squashed commits are merged using the fast-forward option.

I like the explicit notion of a non-fast-forward merge-commit. I also like being able to go through the individual commits in a branch to understand the thought process behind a feature. With fast-forward squash merges, a lot of information is lost.

I'm also against GitHub's "update branch" button, since it doesn't perform a rebase against the target branch, but instead merges it. I suppose this goes well with squash merge, since then all those merge-commits disappear, but without squashing, they make the log history very cluttered.

The way I like to circumvent all of this is to manually rebase, often [interactively](https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d), to make the commit history in a branch sensible and uncluttered.

**Comment by goofballLogic** [comment]
ok well i have to disagree but I'm happy to rebase myself and you'll then need to reapprove

**Comment by goofballLogic** [comment]
At some stage we should perhaps do an audit of the settings we prefer for the repository so you and I (at least) are in sync on this stuff.

**Comment by asbjornu** [comment]
> At some stage we should perhaps do an audit of the settings we prefer for the repository so you and I (at least) are in sync on this stuff.

Definitely! I'm not fuzzy about it, I've just set it up according to my defaults and preference.

**Comment by goofballLogic** [comment]
This will need to be re-approved now I squashed the commits manually

**Pull request merged**